library/core/src/lib.rs

```diff
@@ -93,6 +93,7 @@
 #![warn(missing_debug_implementations)]
 #![warn(missing_docs)]
 #![allow(explicit_outlives_requirements)]
+#![allow(incomplete_features)]
```

Comment:
I am quite concerned about `allow`. That makes it way too easy to use features such as `generic_const_exprs` that are just not ready yet for use in the standard library.

This PR should at least have consulted t-types to ensure that `adt_const_params` is sufficiently stable that it can be used internally in the standard library. Cc @lcnr 

But even then I think we actually want `forbid(incomplete_features)` here to avoid people just adding feature gates when it seems convenient. Or even better, set some flag in bootstrap which ensures that the entire sysroot is built without incomplete features.